// eslint-config-standard-with-typescript/lib/index.js
"use strict";
module.exports = {
    extends: 'eslint-config-standard',
    parserOptions: {
        ecmaFeatures: {
            // https://github.com/standard/eslint-config-standard/issues/95
            jsx: false
        }
    },
    plugins: ['typescript'],
    overrides: [
        {
            files: ['*.ts'],
            rules: {
                'camelcase': ['warn', { properties: 'never' }],
                'no-array-constructor': 'off',
                'no-undef': 'off',
                'no-unused-vars': 'off',
                'no-useless-constructor': 'warn',
                'typescript/adjacent-overload-signatures': 'error',
                'typescript/explicit-function-return-type': 'error',
                'typescript/explicit-member-accessibility': 'error',
                'typescript/member-delimiter-style': ['error', { delimiter: 'none' }],
                'typescript/no-angle-bracket-type-assertion': 'error',
                'typescript/no-array-constructor': 'error',
                'typescript/no-empty-interface': 'error',
                'typescript/no-namespace': 'error',
                'typescript/no-non-null-assertion': 'error',
                'typescript/no-triple-slash-reference': 'error',
                'typescript/no-unused-vars': 'error',
                'typescript/no-use-before-define': ['error', { functions: false, classes: false, variables: false, typedefs: false }],
                'typescript/no-var-requires': 'error',
                'typescript/type-annotation-spacing': 'error'
            }
        }
    ]
};
